### PR TITLE
Make all tests green on JDK 14

### PR DESCRIPTION
**Pull request by rweisleder**
Let's make [#AllTestsGreenOnJDK14](https://twitter.com/mreinhold/status/1243214596854276097) 😉 

@codecholeric: I saw that you started to release v0.14.0. It would be really cool to merge this PR before doing the actual release. You know, v0._14_.0 would be compatible with JDK _14_... 😅

### CODE REVIEW

**Comment by rweisleder** [comment]
Currently, some code in the test utilities fails on JDK 14:
```
java.lang.reflect.InaccessibleObjectException: Unable to make public abstract jdk.internal.PreviewFeature$Feature jdk.internal.PreviewFeature.feature() accessible: module java.base does not "exports jdk.internal" to unnamed module @312b0be6
java.lang.RuntimeException: java.lang.reflect.InaccessibleObjectException: Unable to make public abstract jdk.internal.PreviewFeature$Feature jdk.internal.PreviewFeature.feature() accessible: module java.base does not "exports jdk.internal" to unnamed module @312b0be6
	at com.tngtech.archunit.testutil.TestUtils.invoke(TestUtils.java:41)
	at com.tngtech.archunit.testutil.assertion.JavaAnnotationAssertion.propertiesOf(JavaAnnotationAssertion.java:40)
	at com.tngtech.archunit.testutil.assertion.JavaAnnotationAssertion.propertiesOf(JavaAnnotationAssertion.java:32)
	at com.tngtech.archunit.testutil.assertion.JavaAnnotationAssertion.propertiesOf(JavaAnnotationAssertion.java:26)
	at com.tngtech.archunit.testutil.assertion.JavaMembersAssertion.assertEquivalent(JavaMembersAssertion.java:92)
	at com.tngtech.archunit.testutil.assertion.JavaMemberAssertion.isEquivalentTo(JavaMemberAssertion.java:24)
	at com.tngtech.archunit.testutil.assertion.JavaMembersAssertion.matchInAnyOrder(JavaMembersAssertion.java:56)
	at com.tngtech.archunit.testutil.assertion.JavaMembersAssertion.matchCasted(JavaMembersAssertion.java:86)
	at com.tngtech.archunit.testutil.assertion.JavaMembersAssertion.matchInAnyOrderMembersOf(JavaMembersAssertion.java:79)
	at com.tngtech.archunit.lang.syntax.elements.GivenMembersDeclaredInClassesThatTest.haveSimpleNameStartingWith(GivenMembersDeclaredInClassesThatTest.java:101)
	...
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make public abstract jdk.internal.PreviewFeature$Feature jdk.internal.PreviewFeature.feature() accessible: module java.base does not "exports jdk.internal" to unnamed module @312b0be6
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:349)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:289)
	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:196)
	at java.base/java.lang.reflect.Method.setAccessible(Method.java:190)
	at com.tngtech.archunit.testutil.TestUtils.invoke(TestUtils.java:38)
	... 60 more
```

**Comment by codecholeric** [comment]
Thanks a lot for the initiative :smiley: Unfortunately this release was a pretty tough thing somehow (Gradle publishing change + changes on the classpath scanning), so I just wanted to get it out without any further delays :sweat: But we can simply release soon again and include this. As far as I understand ArchUnit is pretty much compatible with JDK 14 though (I've tried it on JDK 14 projects), only the build of ArchUnit itself does not work yet with JDK 14 (except for the little incompatibility, that the annotation proxy `toString()` might not produce the same?).
Anyway, I'll look into it. I've noticed the thing with `RetentionType(RUNTIME)` as well and I have fixed the same thing on my local branch implementing generic support :sweat_smile:

**Comment by rweisleder** [comment]
> As far as I understand ArchUnit is pretty much compatible with JDK 14 though (I've tried it on JDK 14 projects), only the build of ArchUnit itself does not work yet with JDK 14 (except for the little incompatibility, that the annotation proxy `toString()` might not produce the same?).

Exactly, ArchUnit itself works with JDK 14. The only thing that changed was the `toString()` of annotation proxies.

File: archunit/src/test/java/com/tngtech/archunit/core/PluginLoaderTest.java [hunk 1]
```diff
@@ -1,41 +1,62 @@
...
+
+        System.setProperty("java.version", "13");
+        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(pluginTypeBetweenJava9AndJava13);
+
+        System.setProperty("java.version", "14");
+        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(pluginTypeAfterJava13);
     }
 
     // PluginLoader memoizes the loaded plugin
     private PluginLoader<Object> loadsArrayListForJava9FallbackHashSet() {
```
Comment (hankem) [inline archunit/src/test/java/com/tngtech/archunit/core/PluginLoaderTest.java:48]: Now that this demo-`PluginLoader` returns one of the semantic variables, the name refering specifically to `ArrayList` and `HashSet` (but not `HashMap`) is a bit confusing.

How about naming this method `loadVersionDependentPluginType`?

File: archunit/src/main/java/com/tngtech/archunit/core/domain/AnnotationPropertiesFormatter.java [hunk 2]
```diff
@@ -114,8 +132,21 @@ public String apply(String input) {
             return this;
         }
 
-        AnnotationValueFormatter build() {
-            return new AnnotationValueFormatter(this);
+        /**
+         * Configures that the identifier is omitted if the annotation is a
+         * <a href="https://docs.oracle.com/javase/specs/jls/se13/html/jls-9.html#jls-9.7.3">single-element annotation</a>
```
Comment (hankem) [inline archunit/src/main/java/com/tngtech/archunit/core/domain/AnnotationPropertiesFormatter.java:137]: The PR for JDK14 refers to Java SE13? 😜

Comment (rweisleder) [reply archunit/src/main/java/com/tngtech/archunit/core/domain/AnnotationPropertiesFormatter.java:137]: I tried to be consistent with other references to the JLS like in `JavaClass` 😜 
https://github.com/TNG/ArchUnit/blob/1b3a73fd935032869642ca78a20304040da0340a/archunit/src/main/java/com/tngtech/archunit/core/domain/JavaClass.java#L263-L264

**Comment by rweisleder** [comment]
LGTM!

Additionally, I also renamed that one method in `PluginLoaderTest`.

File: archunit/src/test/java/com/tngtech/archunit/core/PluginLoaderTest.java (hunk 1 above)
Comment (codecholeric) [reply archunit/src/test/java/com/tngtech/archunit/core/PluginLoaderTest.java]: True, I think this should now also refer to the "meta names"...

**Comment by codecholeric** [comment]
Cool, thanks :smile:

File: archunit/src/main/java/com/tngtech/archunit/core/domain/AnnotationPropertiesFormatter.java (hunk 2 above)
Comment (codecholeric) [reply archunit/src/main/java/com/tngtech/archunit/core/domain/AnnotationPropertiesFormatter.java]: I'm afraid this is mostly inconsistent anyway :thinking: Maybe we should at some point update all the links to the current docs... For now, the link for JDK 14 is almost the same, I just used that one.

**Pull request merged**